test/testNestedStructures.py

```diff
@@ -0,0 +1,211 @@
+            if (a != 0)
+               a = b;
+
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
+        self.assertEqual(2, result[1].max_nested_structures)
+
+    def test_nested_loop_mixed_brackets(self):
+        result = process_cpp("""
+        x c() {
+          if (a && b) {
+            if (a != 0)
+              a = b;
+          }
+        }
+        x a() {
+          if (a && b)
+            if (a != 0) {
+               a = b;
+            }
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
+        self.assertEqual(2, result[1].max_nested_structures)
+
+    def test_equal_metric_structures(self):
+        result = process_cpp("""
+        x c() {
+          if (a && b) {
+            if(a != 0){
+                a = b;
+            } else b = a;
+          }
+          if (a && b){
+            if(a != 0)
+                a = b;
+            else
+                b = a;
+          }
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
+
+    def test_while(self):
+        result = process_cpp("""
+        x c() {
+          while (a && b) {
+            if(a != 0){
+                a = b;
+            }
+            while (s) continue;
+          }
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
+
+    def test_do(self):
+        result = process_cpp("""
+        x c() {
+          do {
+            if(a != 0){
+                a = b;
+            } else {
+                b = a;
+            }
+          } while (a && b);
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
+
+    def test_try_catch(self):
+        result = process_cpp("""
+        x c() {
+          try {
+            if (a != 0) {
+                foo();
+            } else {
+                bar();
+            }
+          } catch (auto& err) {}
+        }
+        x a() {
+          try {
+            foo();
+          } catch (auto& err) {
+            if (a != 0) {
+                bar();
+            } else {
+                baz();
+            }
+          }
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
+        self.assertEqual(2, result[1].max_nested_structures)
+
+    def test_switch_case(self):
+        """Switch-Case is one control structure."""
+        result = process_cpp("""
+        x c() {
+          switch (a) {
+            case 1:
+              if (b && c) break;
+            case 0:
+            default:
+              return;
+          }
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
+
+    def test_scope(self):
+        result = process_cpp("""
+        x c() {
+          {{{{{{for (a : c) {
+            {{{{if(a != 0) {{{{
+                a = b;
+            }}}}}}}}
+          }}}}}}}
+        }
+        """)
+        self.assertEqual(2, result[0].max_nested_structures)
```

Comment:
Technically it is not control-flow structure,
but I think it is a good idea to include bare scope into this metric.
However, the bare braces have more to it than control-flow structures.
For example, wrapping your code with openmp pragma to get it parallelized when built with openmp; the code that does the job stays the same for all builds.